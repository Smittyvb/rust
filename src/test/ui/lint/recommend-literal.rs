type Real = double;
//~^ ERROR cannot find type `double` in this scope
//~| HELP you may have meant to use a literal instead

fn main() {
    let x: Real = 3.5;
    let y: long = 74802374902374923;
    //~^ ERROR cannot find type `long` in this scope
    //~| HELP you may have meant to use a literal instead
}

fn z(a: boolean) {
    //~^ ERROR cannot find type `boolean` in this scope
    //~| HELP you may have meant to use a literal instead
}

fn a() -> byte {
//~^ ERROR cannot find type `byte` in this scope
//~| HELP you may have meant to use a literal instead
    3
}

struct Data { //~ HELP you might be missing a type parameter
    width: float,
    //~^ ERROR cannot find type `float` in this scope
    //~| HELP you may have meant to use a literal instead
    depth: Option<int>,
    //~^ ERROR cannot find type `int` in this scope
    //~| HELP you may have meant to use a literal instead
}

trait Stuff {}
impl Stuff for short {}
//~^ ERROR cannot find type `short` in this scope
//~| HELP you may have meant to use a literal instead
