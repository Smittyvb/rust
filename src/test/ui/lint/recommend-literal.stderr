error[E0412]: cannot find type `double` in this scope
  --> $DIR/recommend-literal.rs:1:13
   |
LL | type Real = double;
   |             ^^^^^^
   |             |
   |             not found in this scope
   |             help: try using a Rust type instead: `f64`

error[E0412]: cannot find type `long` in this scope
  --> $DIR/recommend-literal.rs:7:12
   |
LL |     let y: long = 74802374902374923;
   |            ^^^^
   |            |
   |            not found in this scope
   |            help: try using a Rust type instead: `i64`

error[E0412]: cannot find type `boolean` in this scope
  --> $DIR/recommend-literal.rs:12:9
   |
LL | fn z(a: boolean) {
   |         ^^^^^^^
   |         |
   |         not found in this scope
   |         help: try using a Rust type instead: `bool`

error[E0412]: cannot find type `byte` in this scope
  --> $DIR/recommend-literal.rs:17:11
   |
LL | fn a() -> byte {
   |           ^^^^
   |           |
   |           not found in this scope
   |           help: try using a Rust type instead: `u8`

error[E0412]: cannot find type `float` in this scope
  --> $DIR/recommend-literal.rs:24:12
   |
LL |     width: float,
   |            ^^^^^
   |            |
   |            not found in this scope
   |            help: try using a Rust type instead: `f32`

error[E0412]: cannot find type `int` in this scope
  --> $DIR/recommend-literal.rs:27:19
   |
LL |     depth: Option<int>,
   |                   ^^^ not found in this scope
   |
help: try using a Rust type instead
   |
LL |     depth: Option<i32>,
   |                   ^^^
help: you might be missing a type parameter
   |
LL | struct Data<int> {
   |            ^^^^^

error[E0412]: cannot find type `short` in this scope
  --> $DIR/recommend-literal.rs:33:16
   |
LL | impl Stuff for short {}
   |                ^^^^^
   |                |
   |                not found in this scope
   |                help: try using a Rust type instead: `i16`

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0412`.
